### DISCUSSION TITLE
为什么 Linux 和 macOS 不需要碎片整理 - 面向信仰编程 · /whys-the-design-linux-macos-fragmentation

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#646] [Announcements]
https://draveness.me/whys-the-design-linux-macos-fragmentation/

**Comment by yuchanns** [comment]
哈哈，上一篇文章我的留言说对了一部分

**Comment by xindoo** [comment]
1.ssd存储单元使用的是浮栅晶体管，我理解就是电容，用存储的电荷量来表示数据。数据写入时会通过调整电压的方式让电子流入或者流出，这个过程会导致晶体管老化，最终使得晶体管内存储的电荷量越来越不稳定。 
2. hhd内有好多机械结构，所以很容易出现机械损坏，比如盘片在高速旋转中如果发生震动，会导致磁头直接接触盘面，擦伤盘片，不过hhd一般都有坏道检测和屏蔽的机制，轻微的损伤不会影响到使用。

**Comment by lijiang2014** [comment]
1. 为什么固态硬盘的擦写次数有上限？
![ssd unit](https://i0.hdslb.com/bfs/article/60b559add19f4957f7ee0d0bd05d13360d5ac0aa.jpg@1320w_444h.webp)
* SSD的存储单元浮栅晶体管的工作原理是隧道工艺，栅极和沟道之间有一个隧道氧化层，使得断电下电子依然存在在浮栅里，在对浮柵中的电子进行擦除时，需要在控制栅极上施加强负电压并在源极和漏极端子上施加强正电压,利用隧道效应来使得被捕获的电子通过薄氧化层回到隧道,过程中会有部分电子存留在氧化层里，氧化层破坏程度较高时，工作单元失效

2. 机械硬盘在哪些情况下更容易损坏？
* 机械硬盘中有运动组件，因此在工作时的剧烈震动和撞击，以及突然的调电可能会导致物理损伤
* 机械硬盘中的存储依赖磁性材料，因此强的电磁干扰（比如强磁铁），会破坏破坏存储的数据

**Comment by plzhang4321** [comment]
是不是因为linux的伙伴系统~

**Comment by BrianQy** [comment]
所以我经常等我的MacBook内存几乎满了 然后才去清理垃圾 是不是ssd很快要坏掉了
